libchisel/src/lib.rs

```diff
@@ -16,7 +16,8 @@ pub trait ModuleCreator {
 }
 
 pub trait ModuleTranslator {
-    fn translate(self, module: &mut Module) -> Result<bool, String>;
+    fn translate(self, module: &Module) -> Result<Module, String>;
+    fn translate_inplace(self, module: &mut Module) -> Result<bool, String>;
```

Comment:
Why is there a need for a different name?

Reply:
The alternative if you want to overload the method is put it in a different trait.